### PR TITLE
Integrate fine-tuned BERT-based MITRE TTP classifier from Hugging Face

### PR DESCRIPTION
**Pull request by zandanyd**
Fine-tuned a BERT model for multi-label classification of MITRE ATT&CK TTPs using TRAM dataset.

Exported the model as a TorchScript (model_scripted.pt) and uploaded it to Hugging Face Hub.

Updated TramClassifierExtractor to download model and tokenizer directly from the Hugging Face repo.

Extended classification logic to process blog content, extracted keywords, and QnA answers as input chunks.

Added a tram_classifier_node to the pipeline that performs MITRE TTP classification using the Hugging Face model.

Added support for .env variable SKIP_MITRE_CLASSIFICATION to optionally disable classification.

### CODE REVIEW

**Comment by barvhaim** [comment]
@zandanyd do we have `transformers` in the requirements.txt? make sure the libs you are using are installed in the package.. have you tried to run the docker container?

File: backend/extractors/tram_classifier_extractor.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                    label = self.mlb.classes_[idx]
+                    all_preds[label] = all_preds.get(label, 0) + prob
+
+        # Sort by descending probability
+        sorted_preds = sorted(all_preds.items(), key=lambda x: x[1], reverse=True)
+        sorted_labels = [tid for tid, _ in sorted_preds]
+        return self.enrich_ids_with_metadata(sorted_labels)
+
+    def _split_text_into_chunks(self, text, max_tokens=512, stride=256):
+        sentences = blingfire.text_to_sentences(text).split('\n')
```
Comment (barvhaim) [inline backend/extractors/tram_classifier_extractor.py]: we already use `LangChain` so consider using https://pypi.org/project/langchain-text-splitters

Comment (barvhaim) [reply backend/extractors/tram_classifier_extractor.py]: I mean from langchain.text_splitter

File: backend/extractors/tram_classifier_extractor.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        self.threshold = threshold
+        self.keywords = keywords
+        self.qna = qna or []
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+        # === Load model, tokenizer, label binarizer from Hugging Face ===
+        self.model_path = model_repo
+        self.tokenizer = BertTokenizer.from_pretrained(self.model_path)
+
+        scripted_model_path = hf_hub_download(repo_id=model_repo, filename="model_scripted.pt")
```
Comment (barvhaim) [inline backend/extractors/tram_classifier_extractor.py]: Make sure the class is not initialized if the ENV flag is OFF so we don't download it

File: backend/extractors/tram_classifier_extractor.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                            "name": obj.get("name", "Unknown"),
+                            "url": ref.get("url", "")
+                        }
+        return mapping
+
+    def classify(self):
+        chunks = []
+
+        # 1. Split article
+        if len(self.tokenizer.encode(self.article_content, add_special_tokens=False)) > 512:
```
Comment (barvhaim) [inline backend/extractors/tram_classifier_extractor.py]: if the model config is available, you can get the 512 from it instead of hardcoding.. (usually there is a model config json file)

File: cli.py [hunk 4]
```diff
@@ -90,6 +90,25 @@ def _display_results(console: Console, res: dict, url: str):
     else:
         console.print("No IoCs found\n")
 
+    # TRAM MITRE Classification
```
Comment (barvhaim) [inline cli.py]: Enable this part if the ENV var is set

File: backend/pipeline/state.py [hunk 5]
```diff
@@ -10,4 +10,6 @@ class PipelineState(TypedDict):
     qna: List[Dict]
     keywords_found: List[str]
     iocs_found: List[IOC]
+    mitre_attacks: List[Dict]
```
Comment (barvhaim) [inline backend/pipeline/state.py]: rename to mitre_ttps

**Comment by barvhaim** [comment]
@zandanyd looks good, overall comments / checks:
- make sure all required libs are in the requirements.txt
- rename to `mitre_ttps`
- check if you can use text splitters that built in LangChain instead of brining another lib
- check if you can use the transformers to load and use the model instead of PyTorch code, they should have a simpler code with pipelines https://huggingface.co/docs/transformers/en/pipeline_tutorial
- make sure to not loading the model if the ENV var is not set
